interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1047 @@
+not specified it will default to placing a file of the same name as the input file
+with the extension changed to '.tran' in the working directory.
+
+Reordering the reactions will reorder the species inside of each reaction in descending
+order of molecular weight. This can be useful to create uniform mechanisms that can
+later be searched easily outside of Cantera.
+
+Setting overwrite_files to True will allow yaml2ck to delete a prior version of a
+mechanism and write a file with the same name. If it is False then yaml2ck will throw
+an error.
+
+The cantera-validate keyword is a boolean choice to test that the converted 
+Chemkin-formatted mechanism will load back into Cantera.
+
+Supported Thermodynamics: 
+    NASA7
+
+Supported Reaction models: 
+    Elementary, Three-Body, Falloff, Pressure-Dependent-Arrhenius (PLOGs), Chebyshev
+
+Unsupported Thermodynamic Property Models:
+    NASA9, Shomate, constant-Cp
+
+Unsupported Reaction models (not comprehensive): 
+    Chemically-activated, Blowers-Masel, Surface Reactions
+"""
+
+import os
+import sys
+import math
+import re
+import argparse
+from pathlib import PurePath, Path
+from textwrap import fill
+import cantera as ct
+
+# number of calories in 1000 Joules
+CALORIES_CONSTANT = 4184.0
+
+# Conversion from 1 debye to coulomb-meters
+DEBYE_CONVERSION = 3.33564095e-30
+
+
+def Eformat(val, precision=7, exp_digits=3):
+    """
+    Formats numbers in a pretty format for printing
+
+    :param val:
+        Float to be formatted into a str
+    :param precision:
+        How many digits after the decimal
+    :param exp_digits:
+        How many digits the exponent has
+    """
+    if math.isinf(val) and val > 0:
+        return " inf"
+
+    elif math.isinf(val):
+        return "-inf"
+
+    # round to correct number of significant digits
+    if val != 0:
+        val = round(val, precision - int(math.floor(math.log10(abs(val)))) - 1)
+
+    s = f"{val: .{precision}e}"
+
+    mantissa, exponent = s.split("e")
+    exp_digits += 1  # +1 due to sign
+
+    return f"{mantissa}E{int(exponent):+0{exp_digits}d}"
+
+
+def build_header_text(solution, max_width=120):
+    """
+    Creates the header text for all output files
+
+    :param solution:
+        The `solution` object being converted
+    :param max_width:
+        The maximum width of lines before they start to wrap
+    """
+    text = []
+
+    if "description" in solution.input_header:
+        text_wrap = lambda t: fill(
+            t,
+            width=max_width,
+            break_long_words=False,
+            break_on_hyphens=False,
+            initial_indent="! ",
+            subsequent_indent="!   ",
+        )
+
+        text = solution.input_header["description"].splitlines()
+        text = [text_wrap(line) for line in text]
+    else:
+        note = ""
+
+    ct_note = (
+        f"! Chemkin file was converted from a Cantera {ct.__version__} "\
```

Comment:
```suggestion
        f"! Chemkin file was converted from a Cantera {ct.__version__} "
```

The line continuation isn't needed inside the parentheses.